### PR TITLE
port object merger

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

- porting object_merger package
- remove **ANIMAL** label
- support for autoware_auto_perception_msgs

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [pull request guidelines][pull-request-guidelines]
- [x] Code follows [coding guidelines][coding-guidelines]
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: perception/object_recognition/detection/object_merger/src/object_association_merger/node.cpp [hunk 1]
```diff
@@ -67,28 +64,25 @@ void ObjectAssociationMergerNode::objectsCallback(
...
+  for (size_t object0_idx = 0; object0_idx < input_object0_msg->objects.size(); ++object0_idx) {
     if (direct_assignment.find(object0_idx) != direct_assignment.end()) {  // found
       // The one with the higher score will be hired.
       if (
-        input_object1_msg->feature_objects.at(direct_assignment.at(object0_idx))
-          .object.semantic.confidence <
-        input_object0_msg->feature_objects.at(object0_idx).object.semantic.confidence) {
-        output_msg.feature_objects.push_back(input_object0_msg->feature_objects.at(object0_idx));
+        input_object1_msg->objects.at(direct_assignment.at(object0_idx)).existence_probability <
+        input_object0_msg->objects.at(object0_idx).existence_probability) {
```
Comment (satoshi-ota) [inline perception/object_recognition/detection/object_merger/src/object_association_merger/node.cpp:72]: @yukkysaito @yukke42 please check the following modification :pray: 

- confidence -> existence_probability

File: perception/object_recognition/detection/object_merger/src/object_association_merger/data_association/data_association.cpp [hunk 2]
```diff
@@ -26,246 +26,236 @@ DataAssociation::DataAssociation() : score_threshold_(0.1)
...
-    autoware_perception_msgs::msg::Semantic::MOTORBIKE,
-    autoware_perception_msgs::msg::Semantic::BICYCLE) = 1;
+    autoware_auto_perception_msgs::msg::ObjectClassification::MOTORCYCLE,
+    autoware_auto_perception_msgs::msg::ObjectClassification::BICYCLE) = 1;
   can_assign_matrix_(
-    autoware_perception_msgs::msg::Semantic::PEDESTRIAN,
-    autoware_perception_msgs::msg::Semantic::UNKNOWN) = 0;
-  can_assign_matrix_(
-    autoware_perception_msgs::msg::Semantic::ANIMAL,
-    autoware_perception_msgs::msg::Semantic::UNKNOWN) = 0;
```
Comment (satoshi-ota) [inline perception/object_recognition/detection/object_merger/src/object_association_merger/data_association/data_association.cpp:73]: @yukkysaito @yukke42 please check the following modification.

- remove ANIMAL label from following 4 matrixes:
  - can_assing_matrix
  - max_dist_matrix
  - max_area_matrix
  - min_area_matrix

File: perception/object_recognition/detection/object_merger/src/object_association_merger/node.cpp (hunk 1 above)
Comment (yukkysaito) [reply perception/object_recognition/detection/object_merger/src/object_association_merger/node.cpp:72]: LGTM

File: perception/object_recognition/detection/object_merger/src/object_association_merger/data_association/data_association.cpp (hunk 2 above)
Comment (yukkysaito) [reply perception/object_recognition/detection/object_merger/src/object_association_merger/data_association/data_association.cpp:73]: No problem

**Review by yukkysaito — APPROVED** [review]

**Pull request merged**